### PR TITLE
helpers-apostrophies + hard space

### LINKED ISSUE #7185: show name searched for with incorrect spacing characters
I've been trying to debug why this added show "the real housewives of beverly hills" (don't judge me - wife's show) was not being downloaded even though it was on the download provider. It appears that the show name is stored (or just searched?) with some non-alpha characters which causes the search to not retrieve any results. 

For example, with the show added in a normal way, the search appears as below - note the mix of `+`, `%C2%A0` (URL encoded space character in UTF-8 format). The search should be separating words only with `+`. Searches formatted this way return no results:

```
2021-06-18 13:07:37 DEBUG :: SEARCHQUEUE-MANUAL-196741 :: [Jackett (BTN)] :: GET URL: http://10.0.0.3:9117/api/v2.0/indexers/broadcasthenet/results/torznab/api?t=tvsearch&limit=100&offset=0&cat=5000&maxage=500&apikey=XXX&q=The+Real+Housewives+of%C2%A0Beverly%C2%A0Hills+S11E05 [Status: 200]
```

If I add a scene exception using the exact same name as the show, the search is formatted as expected with only `+` as a space seperator. To add the show with the exact same name I have to use caps on the first letter of each word. Now with the exception added, Sickchill is doing both searches, one incorrect (as above) and one correct. The below search returns results and those results are identified correctly in Sickchill:

```
2021-06-18 13:07:35 DEBUG :: SEARCHQUEUE-MANUAL-196741 :: [Jackett (BTN)] :: GET URL: http://10.0.0.3:9117/api/v2.0/indexers/broadcasthenet/results/torznab/api?t=tvsearch&limit=100&offset=0&cat=5000&maxage=500&apikey=XXX&q=The+Real+Housewives+Of+Beverly+Hills+S11E05 [Status: 200]
```

I believe this issue will effect all search types, not just Jackett searches. I was having the same problem using a built-in provider as well - documented in issue #7089. However, using Jackett makes the issue more obvious with the logs providing the GET URL.

### LINKED ISSUE #7190: Auto Scene Exception for ’ not working, being searched
New show added 'Clarkson's Farm' and it automatically has a scene exception 'clarksons farm' but SC only searches for 'Clarkson%E2%80%99s+Farm' the automatically added exception that was expected to also be searched for 'clarksons+farm' isn't searched for.
```
2021-06-21 13:58:38 INFO :: SEARCHQUEUE-MANUAL-378165 :: Beginning manual search for: [Clarkson’s Farm - S01E03 - Shopping]
2021-06-21 13:58:38 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: Building internal name cache for Clarkson’s Farm
2021-06-21 13:58:38 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: Internal name cache for Clarkson’s Farm set to: [ clarksons farm ]
2021-06-21 13:58:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: Search Mode: Episode
2021-06-21 13:58:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: Search String: Clarkson’s Farm S01E03
2021-06-21 13:58:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: GET URL: https://www.torrentday.com/t.json?2=1&26=1&7=1&24=1&34=1&29=1&q=Clarkson%E2%80%99s+Farm+S01E03 [Status: 200]
2021-06-21 13:58:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: Data returned from provider does not contain any torrents
```
Test scene exceptions by adding 'Clarksons farm' to the exceptions (capital C so it's different) and it shows Clarksons+farm being searched for in the debug log. As mentioned above I did expect 'clarkson+farm' from the automatic exception to have been searched for and not require this additional exception.

```
2021-06-21 14:05:32 DEBUG :: WEBSERVER-HOME_0 :: Updating show exceptions
2021-06-21 14:05:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: Building internal name cache for Clarkson’s Farm
2021-06-21 14:05:39 INFO :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: Performing episode search for Clarkson’s Farm
2021-06-21 14:05:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: Search Mode: Episode
2021-06-21 14:05:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: Search String: Clarksons farm S01E03
2021-06-21 14:05:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: GET URL: https://www.torrentday.com/t.json?2=1&26=1&7=1&24=1&34=1&29=1&q=Clarksons+farm+S01E03 [Status: 200]
2021-06-21 14:05:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: Found result: Clarksons Farm S01E03 1080p HEVC x265-MeGusta with 120 seeders and 2 leechers
2021-06-21 14:05:39 DEBUG :: SEARCHQUEUE-MANUAL-378165 :: [TorrentDay] :: Found result: Clarksons Farm S01E03 480p x264-mSD with 73 seeders and 1 leechers
```

### PR DESCRIPTION
**Pull request by BKSteve**
added another bad character to 'if not anime' to resolve #7190 
This is a tiny change and not specifically from develop but from MASTER as develop

added non breaking space re.sub line but can't fully test but should fix #7185

### CODE REVIEW

File: sickchill/oldbeard/helpers.py [hunk 1]
```diff
@@ -708,6 +708,7 @@ def sanitizeSceneName(name, anime=False):
     name = name.replace("&", "and")
     name = re.sub(r"[- /]+", ".", name)
     name = re.sub(r"[.]+", ".", name)
+    name = re.sub(r" ", ".", name)
```
Comment (miigotu) [inline sickchill/oldbeard/helpers.py]: nbsp should be replaced with a regular space

**Comment by BKSteve** [comment]
the character in the first " " is NBSP and should in theory be replaced with a normal space however in this situation where we are converting all characters that are perceived as a separator to . then this is correct.
Also I now understand that ^\S can capture these and is a little cleaner than the "NBSP" character which isn't so clear when looking at the code.
I will look for the point where the GET URL: code is concatenated and see where we can influence the search. That still shows  %C2%A0 so we can search for what is more normal %20.
Grateful for any guidance

**Comment by BKSteve** [comment]
Just to confirm this edit now saves the filename without NBSP in the database and searches too.
filename URLencoded as stored in database 
The%20Real%20Housewives%20of%20Beverly%20Hills
old encode
The%20Real%20Housewives%20of%C2%A0Beverly%C2%A0Hills

```
2021-06-23 17:29:28 DEBUG :: SEARCHQUEUE-MANUAL-196741 :: [TorrentDay] :: GET URL: https://www.torrentday.com/t.json?2=1&26=1&7=1&24=1&34=1&29=1&q=The+Real+Housewives+of+Beverly+Hills+S11E04 [Status: 200]
```

Once this is in master
https://github.com/SickChill/sickchill.github.io/commit/62f76250669cb1b84745781484c8574f73c5ea5e
can be removed.
Below fix neg of neg changed the "^\S" which is "\s"

**Comment by BKSteve** [comment]
The apostrophe fix isn't working properly.

- [ ] track down why allPossibleShowNames doesn't have both with ' and without.
- [ ] look why the internal name cache name isn't being searched for.

**Comment by miigotu** [comment]
If you restart after adding a show, the name cache is built. Just need to add the build name cache call added to the search threads, with a thread lock

**Comment by BKSteve** [comment]
I see in the logs the name cache being built and many shows including the one that's annoying 'Clarkson's Farm' gets changed to it's message [ clarksons farm ] but when it comes time to search the name_cache name doesn't seem to be used for the search but the show_name uncleaned is used. I was expecting to have the clean name searched for or both, but not the uncleaned name.
The learning continues...
PS. after adding shows and modifying code I always restart.
Also tvdb changed the name from having a curved apostrophe to the straight one.

**Pull request merged**